lib/ansible/modules/cloud/amazon/iam_server_certificate_facts.py

```diff
@@ -35,134 +42,158 @@
+                    returned: always
+                    type: str
+                    sample: "/"
+                expiration:
+                    description: The date and time this server certificate will expire, in ISO 8601 format.
+                    returned: always
+                    type: str
+                    sample: "2017-06-15T12:00:00+00:00"
+                upload_date:
+                    description: The date and time this server certificate was uploaded, in ISO 8601 format.
+                    returned: always
+                    type: str
+                    sample: "2015-04-25T00:36:40+00:00"
 '''
 
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
 
+# Non-ansible imports
 try:
-    import boto3
-    import botocore.exceptions
-    HAS_BOTO3 = True
+    import botocore
 except ImportError:
-    HAS_BOTO3 = False
+    pass  # caught by AnsibleAWSModule
 
-from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
 
+def _get_server_certs(connection, module, name, path):
+    """
+    Get one or more IAM server certificates facts based on name or path. If not found, return None.
 
-def get_server_certs(iam, name=None):
-    """Retrieve the attributes of a server certificate if it exists or all certs.
-    Args:
-        iam (botocore.client.IAM): The boto3 iam instance.
-
-    Kwargs:
-        name (str): The name of the server certificate.
-
-    Basic Usage:
-        >>> import boto3
-        >>> iam = boto3.client('iam')
-        >>> name = "server-cert-name"
-        >>> results = get_server_certs(iam, name)
-        {
-            "upload_date": "2015-04-25T00:36:40+00:00",
-            "server_certificate_id": "ADWAJXWTZAXIPIMQHMJPO",
-            "certificate_body": "-----BEGIN CERTIFICATE-----\nbunch of random data\n-----END CERTIFICATE-----",
-            "server_certificate_name": "server-cert-name",
-            "expiration": "2017-06-15T12:00:00+00:00",
-            "path": "/",
-            "arn": "arn:aws:iam::911277865346:server-certificate/server-cert-name"
-        }
+    :param connection: AWS boto3 glue connection
+    :param module: Ansible module
+    :param name: The name of the server certificate
+    :param path: The path of the server certificate
+    :return: IAM server certificates dict or None if not found
     """
-    results = dict()
+
+    response = dict()
+    response['certificates'] = []
+
     try:
         if name:
-            server_certs = [iam.get_server_certificate(ServerCertificateName=name)['ServerCertificate']]
+            response['certificates'].append(connection.get_server_certificate(ServerCertificateName=name))[
+                'ServerCertificate']
+        elif path:
+            server_certs = connection.list_server_certificates(PathPrefix=path)['ServerCertificateMetadataList']
```

Comment:
It would be good if this used pagination, same below (https://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Paginator.ListServerCertificates).